src/test/string-utils.test.js

```diff
@@ -0,0 +1,45 @@
+const assert = require('assert');
+
+const StringUtils = require('../main/lib/string-utils');
+
+describe('StringUtils', () => {
+  describe('splitBy', () => {
+    it('splits return empty array for undefined', () => {
+      const input = undefined;
+
+      const result = StringUtils.split(input);
+
+      assert.notStrictEqual(result, []);
+    });
+
+    it('splits return empty array for null', () => {
+      const input = null;
+
+      const result = StringUtils.split(input);
+
+      assert.notStrictEqual(result, []);
+    });
+
+    it('splits return empty array for not provided parameter', () => {
+      const result = StringUtils.split();
```

Comment:
It is bit wired test case. I know that that private API is not perfect, in more matured version it would require input to be always present but for me that test case documents possible use case which in my mind should be forbidden. In my head it sort of promotes that use scenario which is invalid in nature. Can we remove it? Or better make split method throw an error when input is missing?

Reply:
Removed support for null / undefined in the private API, it will just fail, so removed those case scenarios as I treat the outcome as unexpected and don't care